spec/system/admin/reports/orders_and_distributors_spec.rb

```diff
@@ -0,0 +1,173 @@
+
+              expect(downloaded_filename).to match(/.*\.#{extension}/)
+
+              downloaded_file_txt = load_file_txt(extension, downloaded_filename)
+
+              expect(downloaded_file_txt).to have_text header.join(" ")
+              expect(downloaded_file_txt).to have_text(
+                "By Bike 10 Lovely Street Herndon 20170 UPS Ground", count: 5
+              )
+            end
+          end
+        end
+
+        it_behaves_like "reports generated as", "CSV", "csv"
+        it_behaves_like "reports generated as", "Spreadsheet", "xlsx"
+      end
+    end
+
+    context "as admin" do
+      before do
+        login_as_admin
+        visit report_url
+        run_report
+      end
+
+      context "with two orders on the same day at different times" do
+        let(:completed_at1) { 1500.hours.ago } # 1500 hours in the past
+        let(:completed_at2) { 1700.hours.ago } # 1700 hours in the past
+        let(:datetime_start1) { 1600.hours.ago } # 1600 hours in the past
+        let(:datetime_start2) { 1800.hours.ago } # 1600 hours in the past
+        let(:datetime_end) { 1400.hours.ago } # 1400 hours in the past
+        let!(:order3) {
+          create(:order_ready_to_ship, distributor_id: distributor.id, completed_at: completed_at1)
+        }
+        let!(:order4) {
+          create(:order_ready_to_ship, distributor_id: distributor.id, completed_at: completed_at2)
+        }
+
+        context "applying time/date filters" do
+          it "is precise to time of day, not just date" do
+            # When I generate a customer report
+            # with a timeframe that includes one order but not the other
+            find("input.datepicker").click
+            select_dates_from_daterangepicker datetime_start1, datetime_end
+            find(".shortcut-buttons-flatpickr-button").click # closes flatpickr
+
+            run_report
+            # Then I should see the rows for the first order but not the second
+            # One row per line item - order3 only
+            expect(page).to have_text(distributor.name, count: 6)
+            expect(page).to have_text(order3.email, count: 5)
+
+            # setting a time interval to include both orders
+            find("input.datepicker").click
+            select_dates_from_daterangepicker datetime_start2, Time.zone.now
+
+            run_report
+            # Then I should see the rows for both orders
+            expect(page).to have_text(distributor.name, count: 11)
+            expect(page).to have_text(order3.email, count: 5)
+            expect(page).to have_text(order4.email, count: 5)
+          end
+        end
+
+        context "applying distributor filters" do
+          it "displays line items from the correct distributors" do
+            # for one distributor
+            select2_select distributor.name, from: "q_distributor_id_in"
+            run_report
+            expect(page).to have_content(distributor.name), count: 15
```

Comment:
This doesn't seem to be checking that the other distributor was filtered out.

Reply:
Awesome... Very sharp eye :mag:  The `count:` argument must within the brackets - if we use brackets at all! I've learned this yesterday the hard way, this was not very obvious to me. Thank for pointing it out.